### PR TITLE
Add endpoints for secretsmanager routes (create/list)

### PR DESCRIPTION
**Pull request by fishnix**
* Add functionality to create and list secrets (limited to the configured org).
* Add graceful shutdown for server.

### CODE REVIEW

File: secretsmanager/secrets.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"context"
+
+	"github.com/YaleSpinup/s3-api/apierror"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	log "github.com/sirupsen/logrus"
+)
+
+// CreateSecrets creates a secret in the secretsmanager
```
Comment (damnski) [inline secretsmanager/secrets.go]: s/CreateSecrets/CreateSecret/

File: secretsmanager/secrets.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	out, err := s.Service.CreateSecretWithContext(ctx, input)
+	if err != nil {
+		return nil, ErrCode("failed to create secret", err)
+	}
+
+	return out, nil
+}
+
+// ListSecretsWithFIlter lists all of the secrets with a passed filter function
```
Comment (damnski) [inline secretsmanager/secrets.go]: s/ListSecretsWithFIlter/ListSecretsWithFilter/

File: api/handlers_secrets.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	if !ok {
+		msg := fmt.Sprintf("secretsmanager service not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+
+	secrets, err := smService.ListSecretsWithFilter(r.Context(), func(sec *secretsmanager.SecretListEntry) bool {
+		log.Debugf("checking tags for %s to be sure it's part of the org %s", aws.StringValue(sec.Name), s.org)
+		for _, tag := range sec.Tags {
+			if aws.StringValue(tag.Key) == "yale:org" && aws.StringValue(tag.Value) == s.org {
```
Comment (tenyo) [inline api/handlers_secrets.go]: Methinks this should be a parameter defined in the config

File: api/handlers_secrets.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	}
+
+	input := &secretsmanager.CreateSecretInput{}
+	err := json.NewDecoder(r.Body).Decode(input)
+	if err != nil {
+		msg := fmt.Sprintf("cannot decode body into create secret create input: %s", err)
+		handleError(w, apierror.New(apierror.ErrBadRequest, msg, err))
+		return
+	}
+	input.Tags = append(input.Tags, &secretsmanager.Tag{Key: aws.String("yale:org"), Value: aws.String(s.org)})
```
Comment (tenyo) [inline api/handlers_secrets.go]: same `org`

File: secretsmanager/errors.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	return apierror.New(apierror.ErrInternalError, msg, err)
+	case
+	// ErrCodeLimitExceededException for service response error code
+	// "LimitExceededException".
+	//
+	// The request failed because it would exceed one of the Secrets Manager internal
+	// limits.
+	secretsmanager.ErrCodeLimitExceededException,
+
+		// ErrCodeResourceExistsException for service response error code
```
Comment (tenyo) [inline secretsmanager/errors.go]: 👀

File: secretsmanager/errors.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+	// "ResourceExistsException".
+	//
+	// A resource with the ID you requested already exists.
+	secretsmanager.ErrCodeResourceExistsException:
+			// return a conflict
+			return apierror.New(apierror.ErrConflict, msg, aerr)
+case
```
Comment (tenyo) [inline secretsmanager/errors.go]: 👀

File: secretsmanager/errors.go [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	// The request failed because you did not complete all the prerequisite steps.
+	secretsmanager.ErrCodePreconditionNotMetException:
+	return apierror.New(apierror.ErrBadRequest, msg, aerr)
+	case
+	// ErrCodeResourceNotFoundException for service response error code
+	// "ResourceNotFoundException".
+	//
+	// We can't find the resource that you asked for.
+	secretsmanager.ErrCodeResourceNotFoundException:
+			return apierror.New(apierror.ErrNotFound, msg, aerr)
```
Comment (tenyo) [inline secretsmanager/errors.go]: 👀

File: README.md [hunk 8]
```diff
@@ -152,18 +158,69 @@ To create a `task definition`, just `POST` to the endpoint:
...
+
+| Response Code                 | Definition                      |
+| ----------------------------- | --------------------------------|
+| **200 OK**                    | okay                            |
+| **400 Bad Request**           | badly formed request            |
+| **500 Internal Server Error** | a server error occurred         |
+
+### List secrets
+
+Listing secrets is limited to the secrets that belong to the *org*.
```
Comment (tenyo) [inline README.md]: It's unclear what this `org` is and how it's configured, maybe we should elaborate.

File: config/config.example.json [hunk 9]
```diff
@@ -7,17 +7,20 @@
...
     },
     "spinup": {
       "region": "us-east-1",
       "akid": "xxxxxxxx",
       "secret": "xxxxxxxx",
       "default_sgs": ["sg-zzzzzz"],
       "default_subnets": ["subnet-zzzzzz"],
-      "defaultExecutionRoleArn": "arn:aws:iam::22222222222222:role/ecsTaskExecutionRole"
+      "defaultExecutionRoleArn": "arn:aws:iam::22222222222222:role/ecsTaskExecutionRole",
+      "defaultKmsKeyId": "90909090-8888-7777-6666-545454545454"
```
Comment (tenyo) [inline config/config.example.json]: Not important, but why did we switch from snake case (`default_subnets`) to camel case here 😄

Comment (fishnix) [reply config/config.example.json]: too many devs in the code base? 😆

**Pull request merged**